File: src/JuegoDeRol/versionJava/Guerrero.java
package JuegoDeRol.versionJava;

public class Guerrero extends Personaje {

    public Guerrero(String nombre, int hp, int dmge) {
        super(nombre, hp, dmge);
    }

    public void embestir(Personaje objetivo) {
        System.out.println(super.getNombre() + " va a embestir a " + objetivo.getNombre());
        recibirDamage(super.getDamage() * 2, objetivo);
    }
}

File: src/JuegoDeRol/versionKotlin/Guerrero.kt
package org.example.JuegoDeRol.versionKotlin

class Guerrero(nombre: String, hp: Int, dmge: Int): Personaje(nombre, hp, dmge) {

    fun embestir(objetivo: Personaje){

        println("$nombre va a embestir a ${objetivo.nombre}")
        recibirDamage(damge*2, objetivo)

    }

}